### ISSUE TITLE
Unexpected behaviour when retrying a failing AS3 configuration 

### ISSUE THREAD

**Issue by alonsocamaro** [F5Networks/k8s-bigip-ctlr#3340]
#### Setup Details
 CIS Version : Version: 2.15.0, BuildInfo: azure-4976-a5904e200bf11402611448289d6c864fc7427504
 Build: quay.io/f5networks/k8s-bigip-ctlr-devel:a5904e200bf11402611448289d6c864fc7427504
 Agent Mode: AS3
 Orchestration: OSCP      
 Orchestration Version:       
 Pool Mode: Cluster
 Additional Setup details: OVNKubernetes

#### Description  

When CIS retries an AS3 declaration that is failing as shown next:

```
2024/03/15 10:20:17 [ERROR] [AS3] Raw response from Big-IP: map[code:422 declaration:map[class:ADC controls:map[archiveTimestamp:2024-03-15T10:20:05.521Z class:Controls userAgent:CIS/v2.15.0 OCP/v4.12.20] id:urn:uuid:85626792-9ee7-46bb-8fc8-4ba708cfdc1d label:CIS Declaration remark:Auto-generated by CIS schemaVersion:3.45.0 updateMode:selective] results:[map[code:422 declarationId:urn:uuid:85626792-9ee7-46bb-8fc8-4ba708cfdc1d host:localhost message:declaration failed response:01070022:3: The monitor template /Common/www.sc-twotier.com-checkout was not found. runTime:2529 tenant:sc-twotier]]]
2024/03/15 10:20:17 [ERROR] [AS3] Response from BIG-IP: code: 422 --- tenant:sc-twotier --- message: declaration failed
2024/03/15 10:20:17 [DEBUG] [AS3] Posting failed tenants configuration in 30s seconds
```

if a new configuration is done by removing the offending VirtualServer objects that make use of the missing monitor, the old AS3 configuration is still posted

```
^C
[cloud-user@ocp-provisioner routes-bigip]$ oc -n openshift-ingress get vs
NAME      HOST                 TLSPROFILENAME   HTTPTRAFFIC   IPADDRESS     IPAMLABEL   IPAMVSADDRESS   STATUS   AGE
route-a   www.sc-twotier.com   reencrypt-tls                  10.1.10.104                                        4m27s
```

#### Steps To Reproduce
1) Create a VS A that is fine, Create a VS B that misses a monitor in the BIG-IP
2) Wait the declaration to retry
3) Delete the VS B
4) the behaviour seen is that the retried configuration is the configuration in step 1)

#### Expected Result 

CIS should discard the old configuration

#### Actual Result  

CIS seems to post the old configuration (need to double check)

**Comment by trinaths** [comment]
Created [CONTCNTR-4648] for internal tracking,

**Comment by trinaths** [comment]
Fixed in 2.16.1
